scripts/stack-trace-resolve

```diff
@@ -0,0 +1,164 @@
+#! /bin/bash
+
+
+set -e
+
+
+terminal=
+script_name="$(basename ${0})"
+stack_pattern="stack backtrace:"
+reset_state="waiting_stack"
+executable=
+trace_file=
+default_options="-Cfp"
+options="${default_options}"
+supress=0
+stack_trace_pattern_1="^[[:blank:]]*[0-9]*:[[:blank:]]*0x[[:xdigit:]]*$"
+stack_trace_pattern_2="^[[:blank:]]*[0-9]*:.*\(0x.[[:xdigit:]]*\)$"
+
+detect_term()
+{
+    if [ -t 0 ]; then
+        terminal=1
+    else
+        terminal=0
+    fi
+}
+
+drain()
+{
+    local line=;
+    if [ ${terminal} == 0 ]; then
+        while read -r line; do
+        :
+        done
+    fi
+    exit -1
+}
+
+echo_opt()
+{
+    if [ "${supress}" == "0" ]; then
+        printf %s "${1}"
+	printf "\n"
+    fi
+}
+
+err_print()
+{
+    printf %s "${1}" 1>&2;
+    printf "\n" 1>&2
+}
+
+
+parse_line()
+{
+    local line=$1
+    if [[ $line = *[![:ascii:]]* ]]; then
+        err_print "Non Ascii Characters detected in input\n"
+        usage
+        drain
+    fi
+    line=`printf "%s" "${line}" | awk '{$1=$1};1'`
+
+    if [ "${state}" == "waiting_stack" ]; then
+        echo_opt "${line}"
+        if [ "${line}" == "${stack_pattern}" ]; then
+            state="printing"
+        fi
+    elif  [ "${state}" == "printing" ]; then
+        sr_no=`printf %s "${line}" | awk '{print $1}'`
+        offset=`printf %s "${line}" | awk '{print $NF}'`
+        offset=`printf %s "${offset}" | sed 's/(//g' | sed 's/)//g'`
+        if [[ "${line}" =~ ${stack_trace_pattern_1} || "${line}" =~ ${stack_trace_pattern_2} ]]; then
+            echo "$sr_no `addr2line ${options} -e ${executable} ${offset}`"
+        else
+            state=${reset_state}
+            if [ ${state} != "printing" ]; then
+                parse_line "${line}"
+            else
+                echo_opt "${line}"
+            fi
+        fi
+    fi
+}
+
+parse_input()
+{
+    local line=
+    while IFS='' read -r line || [[ -n "$line" ]]; do
+        parse_line "${line}"
+    done
+}
+
+parse_file() {
+    local line=
+    while IFS='' read -r line || [[ -n "$line" ]]; do
+        parse_line "${line}"
+    done < "$1"
+    check_state
+}
+
+usage() {
+    echo -e\
+        "USAGE:\n"\
+        "\b    ${script_name} [OPTIONS] -e <ELF>\n"\
```

Comment:
Did you mean `\t`?

Reply:
No, its back slash
echo prints a space between 2 strings, \b tries to remove that. I just wanted to avoid repeated echos in the script as well as align the code within the function.

Reply:
Oh. Just use a single multi-line argument.